src/components/DataBrowser.vue

```diff
@@ -62,39 +29,35 @@ export default defineComponent({
         .sort((a, b) => (a.name < b.name ? -1 : 1))
     );
 
-    // --- collection handling --- //
-
-    const collections = computed(() => {
-      const patientItems: Collection[] = patients.value.map((patient) => ({
-        name: patient.name,
-        // namespace key so patient.key doesn't conflict with
-        // the keys from DEFAULT_COLLECTIONS
-        key: `#${patient.key}`,
-      }));
-      if (!patientItems.length) {
-        patientItems.push({
-          name: 'No Patients',
-          key: 'no-patients',
-          disabled: true,
-        });
-      }
-      return [...DEFAULT_COLLECTIONS, ...patientItems];
-    });
-
     const hasAnonymousImages = computed(
       () =>
         imageStore.idList.filter((id) => !(id in dicomStore.imageIDToVolumeKey))
           .length > 0
     );
 
-    const panels = ref<number[]>([0]);
+    const panelKeys = computed(
+      () =>
+        new Set([
+          'sampleData',
+          ...(hasAnonymousImages.value ? ['anonymousData'] : []),
+          ...patients.value.map((study) => study.key),
+        ])
+    );
+
+    const { handlePanelChange, openPanels } = usePanels(panelKeys);
+
+    watch(panelKeys, (newSet, oldSet) => {
+      if (newSet.size > oldSet.size && new Set(openPanels.value).has(0)) {
```

Comment:
`new Set(openPanels.value).has(0)`: if perf is a concern here, this is no better than `openPanels.value.includes(0)` since you have to iterate over the array anyways.

Reply:
Includes is better.  Fixed